packages/peregrine/lib/talons/WishlistPage/useWishlist.js

```diff
@@ -1,20 +1,74 @@
-import { useState } from 'react';
+import { useCallback, useEffect, useState } from 'react';
+import { useLazyQuery } from '@apollo/client';
+import mergeOperations from '../../util/shallowMerge';
+import defaultOperations from './wishlist.gql';
 
 /**
  * @function
- *
+ * @param {String} props.wishlistId The ID of the wishlist this item belongs to
+ * @para  {itemsCount} props.itemsCount The items count fo the list.
+ * @param {Boolean} props.collapsed state of is visable
  * @returns {WishListProps}
  */
-export const useWishlist = () => {
-    const [isOpen, setIsOpen] = useState(true);
+export const useWishlist = (props = {}) => {
+    const { id, itemsCount, collapsed } = props;
+    const operations = mergeOperations(defaultOperations, props.operations);
+
+    const [page, setPage] = useState(1);
+    const [isOpen, setIsOpen] = useState(!collapsed);
+    const [isFetchMore, setIsFetchMore] = useState(false);
+
+    const [fetchWhislistItems, queryResult] = useLazyQuery(
+        operations.getCustomerWhislistItems,
+        {
+            fetchPolicy: 'cache-and-network',
+            nextFetchPolicy: 'cache-first',
+            variables: {
+                id,
+                currentPage: 1
+            }
+        }
+    );
+    const { data, error, loading, fetchMore } = queryResult;
 
     const handleContentToggle = () => {
         setIsOpen(currentValue => !currentValue);
     };
 
+    const onLoadMore = useCallback(async () => {
+        setIsFetchMore(true);
+        const currentPage = page + 1;
+        await fetchMore({
+            variables: {
+                id,
+                currentPage
+            }
+        });
+
+        setPage(currentPage);
+        setIsFetchMore(false);
+    }, [id, fetchMore, page]);
+
+    useEffect(() => {
+        setPage(1);
+        if (itemsCount >= 1 && isOpen === true && !data) {
+            fetchWhislistItems();
+        }
+    }, [itemsCount, isOpen, fetchWhislistItems, data]);
+
+    const items =
```

Comment:
Can you memoize this in a `useMemo`?

Reply:
Duplication of Apollo state can lead to strange issues in the app
also, Kent recommends only using useMemo if it is a complex operation, I would leave it like it here. 

**Reference:
https://kentcdodds.com/blog/usememo-and-usecallback**

Reply:
This is a norm we follow in the codebase. We want the whole codebase to look similar. We always wrap such calculations in a `useMemo` because we don't want them to run on every single render even if the logic is trivial or complex.

Reply:
```es6
const items = useMemo(() => {
     return data && data.customer.wishlist_v2.items_v2.items
            ? data.customer.wishlist_v2.items_v2.items
            : []
}, [data])
```

Reply:
I agree with @larsroettig here; this is derived state and does not need to be memoized. Apollo is already memoizing the result and we can expect this data not to change between renders unless variables change.